vignettes/generic.Rmd

```diff
@@ -0,0 +1,385 @@
+
+Now, let's consider another example of `derive_vars_joined()`. For this we'll need to
+additionally use the `join_vars` argument, which is where the user needs to call out
+any variables from the additional dataframe being used in `filter_join`. Note: If a
+specified variable exists in both dataframes, then the suffix ".join" is added to the
+variable from the additional dataframe.
+
+In this case we want to add a datacut flag to `AE`, only for events occurring up
+to and including this day. So in this case `AE` is the original dataframe and the
+datacut source is the additional dataframe that needs to be joined.
+
+```{r eval=TRUE}
+# Create a unique datacut day for each patient
+datacut <- tribble(
+   ~USUBJID,      ~DCUTDY, ~DCUTFL,
+   "01-701-1111", 5,       "Y",
+   "01-701-1047", 25,      "Y"
+ )
+
+# Join datacut flag to AE only for events up to and including this date
+ae_01 <- ae %>%
+  derive_vars_joined(
+    dataset_add = datacut,
+    by_vars = vars(USUBJID),
+    new_vars = vars(DCUTFL),
+    join_vars = vars(DCUTDY),
+    filter_join = AESTDY <= DCUTDY
+  )
+```
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  dataset = ae_01,
+  display_vars = vars(USUBJID, AEDECOD, AESTDY, DCUTFL)
+)
+```
+
+The `derive_vars_joined()` function could also be used to join the original dataframe
+back with itself, which may be necessary if you're comparing across different observations.
+For example, the below shows how a "nadir" calculation would be achieved which checks
+for the worst value prior up to that observation.
+
+Here is how you would derive the highest severity AE the patient has occurred
+post-baseline up to and excluding the current AE day.
+
+```{r eval=TRUE}
+# Add a numeric version of severity for sorting with severe=1, moderate=2, mild=3
+ae_ext <- ae %>%
+  mutate(TEMP_SEVN = as.integer(factor(AESEV, levels = c("SEVERE","MODERATE","MILD"))))
+
+# Derive nadir severity (AENADSEV)
+ae_02 <- ae_ext %>%
+  derive_vars_joined(
+    dataset_add = ae_ext,
+    filter_add = AESTDY > 0,
+    by_vars = vars(USUBJID),
+    order = vars(TEMP_SEVN),
+    new_vars = vars(AENADSEV = AESEV),
+    join_vars = vars(AESTDY),
+    filter_join = AESTDY.join < AESTDY,
+    mode = "first",
+    check_type = "none"
+  )
+```
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  dataset = ae_02,
+  display_vars = vars(USUBJID, AESEV, AESTDY, AENADSEV)
+)
+```
+
+There exists a further function `derive_var_confirmation_flag()`, which follows
+a similar principle to the "joined" set as explained here. This can be used for whenever
+one set of observations "confirms" another set in a dataframe, i.e. the flag is only
+set for your observation if both the obeservation and some other observation meet both
+the user-specified criteria.
+
+## _extreme_ functions {#extreme}
+
+**Description**\
+
+The "extreme" functions are used for finding the first or last observation
+(from `order` and `mode` arguments) within each by group (`by_vars` argument).
```

Comment:
This is correct but all the merged and joined functions do this as well. I am not sure if we can clearly distinguish "merged" and "extreme". `derive_var_extreme_dt()` could also be named `derive_var_merged_dt()`.

Reply:
hmm... isn't the difference with "extreme" that you can enter multiple different sources via the source objects, whereas "merged" is just for 2 datasets? if so i personally quite like having the extreme functions separate, but maybe i just need to explain that a bit more here to avoid us getting such questions from users. i worry that we may be straying too far into creating one set of functions that do absolutely everything, but because they have so many arguments the users can't comprehend the complexity to know how and when to use them so they'll end up just creating their own. for me seeing a function called "extreme" makes me think of low and high so it's easy to equate to the use case.

Reply:
Yes, `derive_var_extreme_dt()` uses multiple sources but this is not true for other "extreme" functions like `derive_var_extreme_flag()` or `derive_extreme_records()`.
All the "extreme" functions select the highest or lowest value but the "merged" and "joined" functions do this as well.
Therefore it is hard to describe the difference between "extreme" and "merged" functions.
I think cleaning up the admiral functions is a good idea. After that it should be easier to finish the vignette.